src/app/content/highlights/hooks/preventHighlightFromDiscarding.ts

```diff
@@ -0,0 +1,27 @@
+import { getType } from 'typesafe-actions';
+import { callHistoryMethod } from '../../../navigation/actions';
+import { AnyAction, Dispatch } from '../../../types';
+import { Middleware } from '../../../types';
+import { clearFocusedHighlight } from '../actions';
+import showConfirmation from '../components/utils/showConfirmation';
+import * as select from '../selectors';
+
+const actionsToIntercept = new Set<AnyAction['type']>([
+    callHistoryMethod,
+].map(getType));
+
+export default (): Middleware => ({dispatch, getState}) => (next: Dispatch) => async(action: AnyAction) => {
```

Comment:
you can't do this will action hooks, and while with redux middleware you can stop the app from responding to history events, you can't stop the browser for processing them, so that would result in some really weird state.

i was thinking that we would handle navigation events in the links such as the ContentLink component and the contentLinkHandler helper. this doesn't help with the back button, but stopping the back button from functioning is a thing that browsers **really really** don't want you to do, so lets not worry about it for now